test/00_transform/MatMul.cu

```diff
@@ -325,6 +348,58 @@ TYPED_TEST(MatMulTestFloatTypes, MediumRectBatched3DStridedBatch)
   MATX_EXIT_HANDLER();
 }
 
+TYPED_TEST(MatMulTestFloatNonComplexTypes, MixedTypes)
+{
+  using ComplexTypeParam = typename float_to_complex_t<TypeParam>::t;
+  // a -> real, b -> real, c -> complex
```

Comment:
a * b resulting in a complex number should not be valid, should it?

Reply:
Interestingly numpy allows that. (Though I agree that it feels weird).
```python
import numpy as np

a = np.ones((3, 4))
b = np.ones((4, 3))
c = np.ones((3, 3), dtype=np.complex128)

np.matmul(a, b, out=c)
print(c)
```

Do you think we should restrict this case?

Reply:
I'm not entirely sure what the use case for that would be other than you wanting to use `c` as a complex later on, but it causes more overhead to convert both `a` and `b` to complex during the GEMM, rather than just converting `c` to complex when it's needed. I don't have a very strong opinion either way, but it doesn't seem like something we'd want to encourage people to do. I think the usefulness for this feature is mixing real/complex a/b since that's used extremely often in signal processing.

Reply:
Makes sense! Also, `restricting first and then allowing` would be easier than `allowing and then having to restrict it` (for whatever reason).

Will disallow this case. Thanks!